packages/vc-verification/src/revocation-verification.ts

```diff
@@ -1,102 +1,91 @@
+import {
+  CredentialRevoked,
+  StatusListEntryVerification,
+} from '@ew-did-registry/revocation';
 import { verifyCredential } from 'didkit-wasm-node';
-import { StatusPurpose, StatusListCredential } from './models';
-import { VCIssuerVerification } from './issuer-verification-vc';
-import { IssuerResolver, RevokerResolver, CredentialResolver } from '.';
+import { RevokerResolver } from '.';
+import {
+  CredentialNotRevoked,
+  InvalidRevokerType,
+  NoCredentialStatus,
+  NoRevokers,
+  RevokerNotAuthorized,
+} from './errors';
+import { issuerDID } from './models';
+import { CredentialVerification } from './credential-verification';
 
 /**
- * A class to validate if the revocation is valid or not
+ * Provides verification of revocation of EnergyWeb role verifiable credential
  */
 export class RevocationVerification {
-  private _revokerResolver: RevokerResolver;
-  private _credentialResolver: CredentialResolver;
-  private _issuerVerification: VCIssuerVerification;
-
   constructor(
-    issuerResolver: IssuerResolver,
-    revokerResolver: RevokerResolver,
-    credentialResolver: CredentialResolver
-  ) {
-    this._revokerResolver = revokerResolver;
-    this._credentialResolver = credentialResolver;
-    this._issuerVerification = new VCIssuerVerification(
-      credentialResolver,
-      issuerResolver
-    );
-  }
+    private revokerResolver: RevokerResolver,
+    private credentialVerification: CredentialVerification // private credentialResolver: CredentialResolver
+  ) {}
 
   /**
-   * Verifies VC revocation for the given namespace and StatusListCredential
-   * @param namespace for which the revocation was executed
-   * @param credential StatusListCredential to verify
-   * @returns
+   * Verifies that status of the role credential is revoked or suspended by authorized revoker
+   * Revoker authorization must be verified by verifiable credential
+   * @param subject DID of the `role` credential subject
+   * @param role fully qualified role name
+   * @param statusList credential which contains revocation status of `role` credential
    */
-  async verifyRevocation(namespace: string, credential: StatusListCredential) {
-    const revokerDID = credential.issuer;
-    if (
-      !this.isStatusPurposeRevocation(
-        credential.credentialSubject.statusPurpose
-      )
-    ) {
-      return false;
+  async verifyRevocation(
+    subject: string,
+    role: string,
+    statusList: StatusList2021Credential
+  ) {
+    const { issuer: revoker } = statusList;
+    const credential = await this.credentialVerification.verifyEnrolment(
+      subject,
+      role
+    );
+    if (!credential.credentialStatus) {
+      throw new NoCredentialStatus(role);
+    }
+
+    const statusVerifier = new StatusListEntryVerification(verifyCredential);
+    try {
+      await statusVerifier.verifyCredentialStatus(credential.credentialStatus);
+      throw new CredentialNotRevoked(role, subject);
```

Comment:
why?

Reply:
Verification of revocation assumes that credential was revoked. So non revoked credential means failed verification. May be I am wrong. @jrhender could you confirm this?

Reply:
@JGiter Hmm, I think we don't need to have `await this.statusVerifier.verifyCredentialStatus` here.
I think the verification of a revocation of a credential at a higher level (e.g. in ICL) would be something like:
```
const vc: VerifiableCredential<EWFRoleSubject>;
const { isMarkedAsRevoked, statusListCredential } = StatusListEntryVerification.verifyCredentialStatus(vc.credentialStatus);
let revokedByAuthorizedRevoker = false;
if (isMarkedAsRevoked) {
  revokedByAuthorizedRevoker = RevocationVerification.verifyRevocation(statusListCredential);
}
```
So I don't think we need to check the if the original credential is revoked in `RevocationVerification.verifyRevocation` function? Is this how you see it as well @Harasz @nichonien ?

Reply:
The way I see it is that the `statusVerifier.verifyCredentialStatus` should check the revocation status for the given `StatusListEntry` and the `RevocationVerification.verifyRevocation` verifies the revoker's authority and internally calls the `statusVerifier.verifyCredentialStatus`. So ICL probably has to just rely on `RevocationVerification` from `ew-credentials` for complete verification of `StatusListEntry` and revocation verification. This would reduce calls in ICL and would be more accurate (making a single call for complete verification).